src/inspector_agent.cc

```diff
@@ -706,6 +751,50 @@ void Url(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(OneByteString(env->isolate(), url.c_str()));
 }
 
+static void InvokeAsyncTaskFnWithId(void (Agent::*asyncTaskFn)(void*),
+                                    const FunctionCallbackInfo<Value>& args);
+
+static void* GetAsyncTask(intptr_t asyncId);
+
+static void AsyncTaskScheduledWrapper(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  Local<String> taskName = args[0].As<String>();
+  String::Value taskNameValue(taskName);
+  StringView taskNameView(*taskNameValue, taskNameValue.length());
+
+  intptr_t task_id = args[1]->IntegerValue(env->context()).FromJust();
```

Comment:
I'm not sure that on inspector side we'd like to support arbitrary taskId:

On Inspector side we are trying to limit async stack traces memory usage, we dedup frames, stacks and limit maximum amount of async stacks which can be stored at particular moment of time (currently it's 128*1024). We cleanup old stacks first, so it means that at moment when Node.js reports 128*1024 + 1 stack we already collected stack trace with id 1. So it means that 2^31 different async task ids should be enough (actually 2^16 should be enough). Node.js can just use nodejs_id % 2^31 as id for inspector. We can land this CL later when we figure out how we can effectively avoid cleanup of old stacks but it probably would mean that async stack implementation is moved to V8 and enabled by default not only during debugging.